service/matching/tasklist/matcher.go

```diff
@@ -363,13 +363,10 @@ forLoop:
 					tm.scope.RecordTimer(metrics.AsyncMatchLocalPollAfterForwardFailedLatencyPerTaskList, time.Since(startT))
 					return nil
 				case <-childCtx.Done():
-				case <-ctx.Done():
+					attempt++
 					cancel()
-					return fmt.Errorf("failed to dispatch after failing to forward task: %w", ctx.Err())
+					continue forLoop
 				}
```

Comment:
I feel _pretty_ sure we can't make this change - there's no way to escape this loop after this change without having successfully processed a task, and `cancel()` is necessarily a noop since the `childCtx` is already `Done`